src/internal/collection/postgres_collection.go

```diff
@@ -301,20 +326,63 @@ func (c *postgresCollection) list(
 		query += " where " + strings.Join(fields, " and ")
 	}
 
+	afterCondition := func(t SortTarget, ord SortOrder) (string, error) {
+		ts, err := targetToSQL(t)
+		if err != nil {
+			return "", err
+		}
+		args = append(args, afterRow)
+		if len(withFields) > 0 {
+			return fmt.Sprintf(" %s %s $%d", ts, sortOrderOperator(ord), len(args)+1), nil
+		} else {
+			return fmt.Sprintf(" where key %s %s $%d", ts, sortOrderOperator(ord), len(args)+1), nil
+		}
+	}
+
 	if opts.Order != SortNone {
 		if order, err := orderToSQL(opts.Order); err != nil {
-			return err
+			return "", nil, err
 		} else if target, err := targetToSQL(opts.Target); err != nil {
-			return err
+			return "", nil, err
 		} else {
+			if afterRow != nil {
+				ac, err := afterCondition(opts.Target, opts.Order)
+				if err != nil {
+					return "", nil, err
+				}
+				query += ac
+			}
 			query += fmt.Sprintf(" order by %s %s", target, order)
+
+		}
+	} else {
+		if afterRow != nil {
+			ac, err := afterCondition(SortByKey, SortAscend)
+			if err != nil {
+				return "", nil, err
+			}
+			query += ac
 		}
 	}
 
 	if opts.Limit > 0 {
 		query += fmt.Sprintf(" limit %d", opts.Limit)
 	}
+	return query, args, nil
+}
 
+func (c *postgresCollection) list(
```

Comment:
This will not return a consistent view of the database if we have to do multiple queries.

Reply:
right we would have to be okay with that here. @brycemcanally mentioned that's the downside of this approach, but that it should be okay for these operations. What do you think?

Reply:
Still not perfect, but one note is we can guarantee that no two rows will be returned twice within the same stream since the results are always sorted, if a row got added/updated between queries we would not revisit it since it would have already been passed in the sort order. That is only with the exception of `SortOrder==ASC && TargetColumn==UPDATED_AT`